rotkehlchen/assets/utils.py

```diff
@@ -71,6 +71,14 @@ def get_or_create_evm_token(
     try:
         ethereum_token = EvmToken(identifier, form_with_incomplete_data=form_with_incomplete_data)
     except (UnknownAsset, DeserializationError):
+        try:
+            Asset(identifier).check_existence()
+            # If exists, delete the asset. Otherwise we will have a db error.
+            # Can happen if the asset is not an evm token, or if name, symbol or decimals is None
+            # while form_with_incomplete_data is False.
+            GlobalDBHandler().delete_asset_by_identifier(identifier)
```

Comment:
So for me to understand what you did is delete the token if we get `UnknownAsset` but this only happens when
```
if missing_basic_data and form_with_incomplete_data is False:
```

or otherwise you are not doing any deletion since the asset was not in the database. 

In the error case the asset had all the information so it was not a problem of missing information. Also seems very dangerous to delete the asset since there could be balances in the user db depending on it and you could lead to an inconsistent state